.github/workflows/deploy.yml

```diff
@@ -0,0 +1,122 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ beta, main ]
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Read node version (package.json)
+        run: echo "::set-output name=version::$(jq -r .engines.node package.json)"
+        id: node-version
+
+      - name: Use Node.js (package.json)
+        uses: actions/setup-node@v1
+        with:
+          node-version: "${{ steps.node-version.outputs.version }}"
+
+      - name: Read `yarn cache dir`
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Load yarn cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Build
+        run: yarn build
+        env:
+          NODE_ENV: production
+
+      - name: Prepare artifact
+        run: |
+          tar -czf fr-api.tar.gz \
+            dist \
+            migrations \
+            schema \
+            static \
+            tools \
+            views \
+            .sequelizerc \
+            build.json \
+            database_example.json \
+            LICENSE.md \
+            localisations.json \
+            package.json \
+            permissions.json \
+            README.md \
+            yarn.lock
+
+      - name: Publish artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: fr-api
+          path: fr-api.tar.gz
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: publish
+
+    env:
+      NODE_ENV: production
+
+    steps:
+      - name: Download artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: fr-api
+
+      - name: Extract artifact
+        run: |
+          tar -xzf fr-api.tar.gz
+          rm fr-api.tar.gz
+
+      - name: Read node version (package.json)
+        run: echo "::set-output name=version::$(jq -r .engines.node package.json)"
+        id: node-version
+
+      - name: Use Node.js (package.json)
+        uses: actions/setup-node@v1
+        with:
+          node-version: "${{ steps.node-version.outputs.version }}"
+
+      - name: Read `yarn cache dir`
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Load yarn cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
```

Comment:
On fuelrats.com, I've elected to avoid bundling `node_modules` into the build artifact in favor of installing again in the deployment step. I've copied that behavior here. This install, unlike the build job, will not include `devDependencies` since `NODE_ENV=production` is set on the whole job (see [L77](https://github.com/FuelRats/api.fuelrats.com/pull/579/files#diff-3680803a67ea02b9d6b8b783f532a8ceR77)). This tends to shave down install and transfer time quite a bit.